### DISCUSSION TITLE
FlatBuffers CMake does not contain version information

### DISCUSSION THREAD

**Discussion by dominiklohmann** [Homebrew/discussions#150] [Everyday usage] [answered]
FlatBuffers installed via Homebrew does not set the version information in the bundled CMake package configuration file as expected, causing `find_package(Flatbuffers 1.12)` to fail.

```
❯ cat /usr/local/lib/cmake/flatbuffers/FlatbuffersConfigVersion.cmake
```

```cmake
set(PACKAGE_VERSION "..")

# Check whether the requested PACKAGE_FIND_VERSION is compatible
if("${PACKAGE_VERSION}" VERSION_LESS "${PACKAGE_FIND_VERSION}")
  set(PACKAGE_VERSION_COMPATIBLE FALSE)
else()
  set(PACKAGE_VERSION_COMPATIBLE TRUE)
  if ("${PACKAGE_VERSION}" VERSION_EQUAL "${PACKAGE_FIND_VERSION}")
    set(PACKAGE_VERSION_EXACT TRUE)
  endif()
endif()
```

This happens because [FlatBuffers uses `git describe` to generate the version components](https://github.com/google/flatbuffers/blob/6d0aae73cd2ad3f8ee93c952af6b33e99049ad9c/CMake/Version.cmake). 

There are two ways to fix this:
- Submit a patch to FlatBuffers, and wait for 1.13 to be released.
- Submit a patch to homebrew-core, that adds `git fetch --all --tags` in the install section of the formula.

I'd strongly prefer to do the latter, but is that even allowed? I couldn't find much information on this and could not find any other formula running `git fetch`.

**Comment by gromgit** [comment, marked as answer]
Patching the source independent of upstream is generally discouraged, but I don't think your solution would work anyway, since the formula is built with the release tarball, which has zero Git metadata.

Better to submit a patch to upstream, and let them fix it for everyone.
